subprojects/ear/src/main/java/org/gradle/plugins/ear/EarPlugin.java

```diff
@@ -193,14 +193,16 @@ public DeploymentDescriptor call() throws Exception {
     }
 
     private void configureConfigurations(final Project project) {
-
+        // Currently 'deploy' and 'earlib' are both _resolvable_ and _consumable_.
+        // In the future, it would be good to split these so that the attributes can differ.
+        // Then 'jvmPluginServices.configureAsRuntimeClasspath()' may be used to configure the resolving configurations.
         ConfigurationContainer configurations = project.getConfigurations();
         Configuration moduleConfiguration = configurations.create(DEPLOY_CONFIGURATION_NAME).setVisible(false)
             .setTransitive(false).setDescription("Classpath for deployable modules, not transitive.");
-        jvmPluginServices.configureAsRuntimeClasspath(moduleConfiguration);
+        jvmPluginServices.configureAttributes(moduleConfiguration, details -> details.library().runtimeUsage().withExternalDependencies());
```

Comment:
Why is this configuration not carrying the new attribute?

Reply:
That's because of the code comment I added a bit further up:

```
// Currently 'deploy' and 'earlib' are both _resolvable_ and _consumable_.
// In the future, it would be good to split these so that the attributes can differ.
// Then 'jvmPluginServices.configureAsRuntimeClasspath()' may be used to configure the resolving configurations
```

This is both "resolvable" and "consumable". And because we don't want to change any consumable defaults (see next comment) we do it like this.

I also think it's dangerous to use `configureAsRuntimeClasspath` here, since this is not only a classpath.

On the long run, this should be split into two configurations.